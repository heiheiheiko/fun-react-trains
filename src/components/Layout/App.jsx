import React from 'react';
import { BrowserRouter as Router, Route, NavLink } from "react-router-dom";
import pages from "../../config/Pages";
import AppAside from "./AppAside";

function App(){
  return (
    <Router>
      <div className="overlay">
        <nav className="overlay__layer mt-small py-small px-normal has-background-light">
          {pages.map((navItem) =>
            <NavLink 
              className="ml-normal"
              to={navItem.path} 
              key={navItem.id} 
              activeClassName="has-text-primary"> 
              {navItem.title} 
            </NavLink>  
          )}
        </nav>
        
        {pages.map((navItem) =>
          <Route path={navItem.path} render={() => navItem.component} key={navItem.id}/>
        )}

        <AppAside/>
      </div>
    </Router>
  );
}

export default App;
