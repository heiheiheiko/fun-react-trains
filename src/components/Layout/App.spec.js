import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecIntelContainer';

import App from './App';

jest.mock('./AppAside', () => ()=> "AppAside");

describe('<App>', () => {
  it('could be renderd', () => {
    const mountedSpecContainer = mount(<SpecContainer><App/></SpecContainer>);
    expect(mountedSpecContainer.find(App).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const specContainer = <SpecContainer><App/></SpecContainer>;
    const renderedValue =  renderer.create(specContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});