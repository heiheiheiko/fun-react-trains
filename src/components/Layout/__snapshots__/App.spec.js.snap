// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> matched the snapshot 1`] = `
<div>
   
  <div
    className="overlay"
  >
    <nav
      className="overlay__layer mt-small py-small px-normal has-background-light"
    >
      <a
        aria-current={null}
        className="ml-normal"
        href="/world"
        onClick={[Function]}
      >
        <span>
          World
        </span>
      </a>
      <a
        aria-current={null}
        className="ml-normal"
        href="/finnland"
        onClick={[Function]}
      >
        <span>
          Finnland
        </span>
      </a>
    </nav>
    AppAside
  </div>
   
</div>
`;
